lib/rules/logical-assignment-operators.js

```diff
@@ -0,0 +1,465 @@
+                        const assignmentOperatorToken = getOperatorToken(assignment);
+
+                        // -> foo ||= foo || bar
+                        yield ruleFixer.insertTextBefore(assignmentOperatorToken, assignment.right.operator);
+
+                        // -> foo ||= bar
+                        yield ruleFixer.removeRange([assignment.right.range[0], assignment.right.right.range[0]]);
+
+                        if (astUtils.isParenthesised(sourceCode, assignment.right.right)) {
+
+                            // Opening parenthesis already removed
+                            const closingParenthesis = sourceCode.getTokenAfter(assignment.right.right);
+
+                            yield ruleFixer.remove(closingParenthesis);
+                        }
+                    }
+                };
+
+                context.report(createConditionalFixer(descriptor, suggestion, cannotBeGetter(assignment.left)));
+            },
+
+            // foo || (foo = bar)
+            'LogicalExpression[right.type="AssignmentExpression"][right.operator="="]'(logical) {
+                if (astUtils.isSameReference(logical.left, logical.right.left)) {
+                    const descriptor = {
+                        messageId: "logical",
+                        node: logical,
+                        data: { operator: logical.operator }
+                    };
+                    const suggestion = {
+                        messageId: "convertLogical",
+                        data: { operator: logical.operator },
+                        *fix(ruleFixer) {
+                            if (sourceCode.getCommentsInside(logical).length > 0) {
+                                return;
+                            }
+
+                            const requiresParenthesis = logical.parent.type !== "ExpressionStatement" &&
+                                                        (astUtils.getPrecedence({ type: "AssignmentExpression" }) < astUtils.getPrecedence(logical.parent));
+
+                            if (!astUtils.isParenthesised(sourceCode, logical) && requiresParenthesis) {
+                                yield ruleFixer.insertTextBefore(logical, "(");
+                                yield ruleFixer.insertTextAfter(logical, ")");
+                            }
+
+                            // Right side has to be parenthesized, otherwise would be parsed as (foo || foo) = bar which is illegal
+                            const leftToStartRight = [logical.range[0], logical.right.range[0]];
+
+                            yield ruleFixer.removeRange(leftToStartRight); // -> foo = bar)
+
+                            const endParenthesis = sourceCode.getTokenAfter(logical.right);
+
+                            yield ruleFixer.remove(endParenthesis); // -> foo = bar
+
+                            const operatorToken = getOperatorToken(logical.right);
```

Comment:
This one also might not work well with mutliple parentheses:

```js
/*eslint logical-assignment-operators: ["error", "always"]*/

a || ((a = b))

```

After the fix:

```js
/*eslint logical-assignment-operators: ["error", "always"]*/

a ||= b)

```

Maybe we could do `yield ruleFixer.removeRange([logical.right.range[1], logical.range[1]]);` instead of removing just one parenthesis.